src/mixins/nr-affiliation-mixin.ts

```diff
@@ -1,68 +1,80 @@
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Mixins } from 'vue-property-decorator'
 import { Action } from 'vuex-class'
 import AuthServices from '@/services/auth.services'
 import BusinessServices from '@/services/business.services'
-import { BusinessRequest, CreateNRAffiliationRequestBody, NameRequestI } from '@/interfaces'
+import { BusinessRequest, NameRequestI } from '@/interfaces'
 import { ActionBindingIF } from '@/interfaces/store-interfaces'
 import { navigate } from '@/plugins'
+import { CommonMixin } from '@/mixins'
+import { CREATED, BAD_REQUEST } from 'http-status-codes'
 
 @Component({})
-export class NrAffiliationMixin extends Vue {
+export class NrAffiliationMixin extends Mixins(CommonMixin) {
   // Global action
   @Action setAffiliationErrorModalVisible!: ActionBindingIF
 
   /**
-   * Creates an affiliation between the current authenticated account and the current NR.
+   * Creates an affiliation between the current authenticated account and the current NR
+   * and then navigates to My Business Registry page.
    * @param nr the NR to affiliate
    */
   async createAffiliation (nr: NameRequestI): Promise<any> {
     try {
       // show spinner since the network calls below can take a few seconds
       this.$root.$emit('showSpinner', true)
 
-      const currentOrganizationId = JSON.parse(sessionStorage.getItem('CURRENT_ACCOUNT')).id
-      const requestBody: CreateNRAffiliationRequestBody = {
-        businessIdentifier: nr.nrNum,
-        phone: nr.applicants?.phoneNumber || '',
-        email: nr.applicants?.emailAddress || ''
+      // NB: fall back is user's default account
+      const accountId = JSON.parse(sessionStorage.getItem('CURRENT_ACCOUNT')).id || 0
+
+      //
+      // try to affiliate NR to current account
+      //
+      const affiliationResponse = await AuthServices.createNrAffiliation(accountId, nr)
+      if (
+        affiliationResponse.status === BAD_REQUEST &&
+        affiliationResponse.data.code === 'NR_CONSUMED'
+      ) {
+        throw new Error('The name request is already affiliated')
```

Comment:
Possible future improvement: if the NR is already affiliated, try to create the business; if the business already exists, go to its dashboard.

Implementing this would require API changes, since it will need to provide the Business Identifier if already affiliated (otherwise there is no way to get it).

Or, we could navigate to the user's Business Registry page, and they could open the business from there. (future)

Reply:
Code is updated. This conversation is obsolete.